paddle/fluid/operators/listen_and_serv_op.cc

```diff
@@ -169,15 +201,85 @@ void ListenAndServOp::RunSyncLoop(framework::Executor *executor,
   }  // while(true)
 }
 
+void ListenAndServOp::RunAsyncLoop(framework::Executor *executor,
+                                   framework::ProgramDesc *program,
+                                   framework::Scope *recv_scope,
+                                   framework::BlockDesc *prefetch_block) const {
+  VLOG(3) << "RunAsyncLoop in";
+  // grad name to block id
+  std::unordered_map<std::string, int32_t> grad_to_id;
+  std::unordered_map<int32_t, std::string> id_to_grad;
+
+  auto grad_to_id_str = Attr<std::vector<std::string>>("grad_to_id");
```

Comment:
`grad_to_id_str` can be generated by `listen_and_serv_op` when initializing, read the `ProgramDesc` blocks and create a mapping, so we can save this attribute.

Reply:
I think grad_to_id_str should be created in Python by transpiler because the transpile logic know how to split the operator and block,  listen_and_serv_op just use the result is fine, or it has to understand the detailed logic of transpiler.

Reply:
Firstly we want to make `listen_and_serv_op` general, that means it should not know that attributes and inputs are "grads" or parameters, it should simply receive the data and run a block.

In that case, for Async Execution, `listen_and_serv` is responsible to determine which block need to run when the data arrives. Just open for discussion.

Reply:
After discussing with @typhoonzero, I get the point, I totally agree with the idea that listen_and_serv_op should be a general operator! We will find a better way to implement async update in the future PRs.